### PR TITLE
Add functions for reindexing

### LINKED ISSUE #47: Create and document a full rehydration, retransformation, and reindexing process
## Acceptance Criteria

- [ ] There's a way to rehydrate (resulting in retransformation + reindexing) using the same cache_version
- [ ] There's a way to to retransform (resulting in reindexing) using the same cache_version
- [ ] There's a way to reindex using the same cache_version
- [ ] Documentation on how to do the above.

# First Step

Read the code implemented for #30

### PR DESCRIPTION
**Pull request by eliotjordan**
Adds functions for deleting processor marker and restarting the producer (clearing state) for each indexing stage.

closes #47

### CODE REVIEW

File: lib/dpul_collections/utilities.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def reindex_hydration_cache(cache_version) do
+    hydration_processor_marker =
+      IndexingPipeline.get_processor_marker!("hydrator", cache_version)
+
+    IndexingPipeline.delete_processor_marker(hydration_processor_marker)
+    GenServer.stop(Figgy.HydrationConsumer)
+  end
+
+  def reindex_all(cache_version) do
```
Comment (tpendragon) [inline lib/dpul_collections/utilities.ex:31]: Do we need to reset the state of everything if want to reindex? Like if we want a full re-hydration, can't we just reset the hydrator? And if we want a full retransformation without hydration, we just reset the transformer?

File: lib/dpul_collections/utilities.ex [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    # This resets the state and last queried marker.
+    GenServer.stop(Figgy.IndexingConsumer)
+  end
+
+  def reindex_transformation_cache(cache_version) do
+    transformation_processor_marker =
+      IndexingPipeline.get_processor_marker!("figgy_transformer", cache_version)
+
+    IndexingPipeline.delete_processor_marker(transformation_processor_marker)
+    GenServer.stop(Figgy.TransformationConsumer)
```
Comment (tpendragon) [inline lib/dpul_collections/utilities.ex:20]: Should we use Broadway.stop?

File: lib/dpul_collections/utilities.ex [hunk 3]
```diff
@@ -0,0 +1,36 @@
+defmodule DpulCollections.Utilities do
+  alias DpulCollections.IndexingPipeline
+  alias DpulCollections.IndexingPipeline.Figgy
+
+  def reindex_solr(cache_version) do
```
Comment (tpendragon) [inline lib/dpul_collections/utilities.ex:5]: Should these functions be part of the Consumer namespace? So like, `Figgy.IndexingConsumer.Restart()` or `Reset()` or `ResetTo(nil)` or something?

**Review by tpendragon — DISMISSED** [review]
More discussion points than specific changes.

**Comment by tpendragon** [comment]
Update: I tried doing it this way while I was reindexing for description and it didn't work - if indexing is running, it writes the marker between me updating and restarting the node, no matter how quickly I went. I think we need to stop the producer in-band somehow.

**Comment by hackartisan** [comment]
> Update: I tried doing it this way while I was reindexing for description and it didn't work - if indexing is running, it writes the marker between me updating and restarting the node, no matter how quickly I went. I think we need to stop the producer in-band somehow.

I feel like we should be able to start these broadway pipelines as named processes so that we can shut them down by referencing them with an atom instead of needing a PID

File: test/dpul_collections/indexing_pipeline/integration/full_integration_test.exs [hunk 4]
```diff
@@ -1,10 +1,10 @@
 defmodule DpulCollections.IndexingPipeline.FiggyFullIntegrationTest do
+  alias ElixirLS.LanguageServer.Providers.CodeLens.Test
```
Comment (hackartisan) [inline test/dpul_collections/indexing_pipeline/integration/full_integration_test.exs:2]: Remove this?

**Review by hackartisan — CHANGES_REQUESTED** [review]

**Review by hackartisan — DISMISSED** [review]

**Review by eliotjordan — COMMENTED** [review]
Nice work. 💯

**Pull request merged**